bootstrap/pkg/kfupgrade/kfupgrade.go

```diff
@@ -217,13 +229,23 @@ func MergeKfDef(oldKfDef *kfdefsv3.KfDef, newKfDef *kfdefsv3.KfDef) {
 }
 
 func (upgrader *KfUpgrader) Generate() error {
-	kfApp := kustomize.GetKfApp(upgrader.NewKfDef)
+	kfApp, err := coordinator.BuildKfAppFromURI(upgrader.TargetPath)
```

Comment:
I think you want to use NewLoadKfAppFromURI.

BuildKFAppFromURI is calling init and generate if you need those methods just call them yourself on the kfapp returned by NewLoadKfAppFromUri.

Reply:
You are already calling Generate below.